hypervisor/arch/x86/vtd.c

```diff
@@ -992,12 +992,16 @@ static void do_action_for_iommus(void (*action)(struct dmar_drhd_rt *))
 	struct dmar_drhd_rt *dmar_unit;
 	uint32_t i;
 
-	for (i = 0U; i < info->drhd_count; i++) {
-		dmar_unit = &dmar_drhd_units[i];
-		if (!dmar_unit->drhd->ignore) {
-			action(dmar_unit);
-		} else {
-			dev_dbg(ACRN_DBG_IOMMU, "ignore dmar_unit @0x%x", dmar_unit->drhd->reg_base_addr);
+	if (info == NULL) {
+		pr_fatal("%s: can't find dmar info\n", __func__);
```

Comment:
can we move this branch to be after next (info!=NULL)?
This is never executed single line code, putting later seems to be better, in case we want to delete it later on.